### PR TITLE
feat: [IOCOM-270] FIMS cookie clearing when leaving the FIMS webview

### PR DESCRIPTION
**Pull request by Vangaorth**
## Short description
This PR removes the FIMS token when leaving the FIMS webview.

## List of changes proposed in this pull request
- Added the proper wrapping method to the cookie manager
- `FimsWebViewScreen` refactored in order to access both domain and FIMS cookie name, caching unchanged data and avoiding unnecessary re-renderings

## How to test
Using the `io-dev-api-server`, start a FIMS flow and check that the FIMS cookie is properly set

### CODE REVIEW

File: ts/utils/cookieManager.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import CookieManager, { Cookie } from "@react-native-cookies/cookies";
 
-export const ioClearCookie = (cb: () => void) => {
-  CookieManager.clearAll().catch(cb);
-};
+export const ioClearCookie = (cb: () => void) =>
+  void CookieManager.clearAll().catch(cb);
+
+export const clearCookie = (url: string, name: string, cb?: () => void) =>
+  void CookieManager.clearByName(url, name).catch(_ => cb?.());
```
Comment (adelloste) [inline ts/utils/cookieManager.ts:7]: `clearByName` is only available on iOS, we should remove cookies on Android too.

File: ts/utils/cookieManager.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
 import CookieManager, { Cookie } from "@react-native-cookies/cookies";
 
-export const ioClearCookie = (cb: () => void) => {
-  CookieManager.clearAll().catch(cb);
-};
+export const ioClearCookie = (cb: () => void) =>
+  void CookieManager.clearAll().catch(cb);
```
Comment (adelloste) [inline ts/utils/cookieManager.ts:4]: To get the same style, could it make sense to rename this function by removing the `io` prefix?

File: ts/utils/cookieManager.ts (hunk 1 above)
Comment (Vangaorth) [reply ts/utils/cookieManager.ts:7]: Reading the library documentation, it seems that there is not a single API that is callable on both platforms in order to remove a previously set cookie. So, what do you say if, upon `clearCookie` invocation, we set the same cookie again, with both an invalid value (for the FIMS' token) and an hardcoded expired date (like `Fri, 1 Jan 2010 00:00:00 -0000`) ?

File: ts/utils/cookieManager.ts (hunk 2 above)
Comment (Vangaorth) [reply ts/utils/cookieManager.ts:4]: Sounds good

File: ts/utils/cookieManager.ts (hunk 1 above)
Comment (Vangaorth) [reply ts/utils/cookieManager.ts:7]: Since it is not working properly, let us switch to `removeSessionCookies`. What do you think?

Comment (adelloste) [reply ts/utils/cookieManager.ts]: Ok, agreed

**Review by adelloste — APPROVED** [review]
LGTM 🏈

**Pull request merged**